### ISSUE TITLE
Exception: missing initial moveto in path definition in 0.2.0-m6

### ISSUE THREAD

**Issue by ozen** [qupath/qupath#376]
Hi Pete,

Thanks for your great work. I've just upgraded to 0.2.0-m6 from m5 and got the following exception when loading an image in a project that worked well in m5:

"QuPath exception: missing initial moveto in path definition"

The project was created using an earlier version, but was working fine in 0.2.0-m5. The exception is thrown for images without any objects as well.

OS: Windows 10, 1903

**Comment by Svidro** [comment]
The milestone versions aren't really intended to be completely inter-compatible. I would recommend creating a new project and importing any scripts or classifiers you might have had into the new project, rather than trying to resuse it. Especially if you are using images without any objects.

That mostly consists of copying the folders named things like "classifiers" and "scripts" into the new project folder.

**Comment by petebankhead** [comment]
Hi @ozen 

The error message sounds like it is connected to drawing shapes, but to further diagnose the problem I would need the original .qpdata files - or at least the full error trace (usually under _View &rarr; Show log_).

However as @Svidro says, the milestones aren't/can't be fully inter-compatible, and there can be no guarantee that an m5 project will work in m6 (although any that I've tried myself have worked ok).

I've written about some of the major changes in v0.2.0-m6 [here](https://petebankhead.github.io/qupath/2019/11/17/sixth-milestone.html). I don't know why these would give an error in your case, but really I'm not able to do anything about it without further information and the ability to reproduce the problem myself.

Can you reproduce the issue using a project created entirely in m6, and describe the steps required?

**Comment by ozen** [comment]
@Svidro I'm not trying to solve my problem, but reporting a potential bug. I'm fine with using 0.2.0-m5. I've already read the sentence you quoted, also the sentence right after it:

> If you find bugs, please report them on GitHub or image.sc - don’t just stick with an older milestone that seemed to work!

And this one under object hierarchy section:

> Things that worked before should still work (if they don’t, please report the bug!)

To reproduce:

1. Create a project in m5, import an image
2. Copy this qpdata to the project entry: [data.zip](https://github.com/qupath/qupath/files/3857086/data.zip)
3. Open project with m6

I don't recall the version that created this qpdata file, but it works with m5. It contains some non-hierarchical annotations.

Stack trace:

> INFO: Bio-Formats version 6.3.0
INFO: Loaded extension Bio-Formats server options (Bio-Formats 6.3.0) (27 ms)
INFO: Loaded extension Experimental commands (26 ms)
INFO: Loaded extension ImageJ extension (90 ms)
INFO: Loaded extension JPen extension (34 ms)
INFO: Loaded extension OpenCV extensions (4 ms)
INFO: Loaded extension Rich script editor extension (562 ms)
INFO: OpenSlide version 3.4.1
INFO: Selected style: null
INFO: Performing update check...
WARN: No changelog found - will not check for updates
INFO: Starting QuPath with parameters: []
INFO: Project set to Project: deneme-project
WARN: Openslide: Property 'openslide.mpp-x' not available, will return default value NaN
WARN: Openslide: Property 'openslide.mpp-y' not available, will return default value NaN
WARN: Openslide: Property 'openslide.objective-power' not available, will return default value NaN
INFO: Image data set to ImageData: Not set, B-11285-15_2.tif
ERROR: QuPath exception: missing initial moveto in path definition
    at java.desktop/java.awt.geom.Path2D$Float.needRoom(Unknown Source)
    at java.desktop/java.awt.geom.Path2D.closePath(Unknown Source)
    at qupath.lib.roi.ShapeSimplifier.simplifyPath(ShapeSimplifier.java:310)
    at qupath.lib.gui.viewer.PathHierarchyPaintingHelper$ShapeProvider.simplifyByDownsample(PathHierarchyPaintingHelper.java:542)
    at qupath.lib.gui.viewer.PathHierarchyPaintingHelper$ShapeProvider.getShape(PathHierarchyPaintingHelper.java:584)
    at qupath.lib.gui.viewer.PathHierarchyPaintingHelper.paintROI(PathHierarchyPaintingHelper.java:434)
    at qupath.lib.gui.viewer.PathHierarchyPaintingHelper.paintObject(PathHierarchyPaintingHelper.java:400)
    at qupath.lib.gui.viewer.PathHierarchyPaintingHelper.paintSpecifiedObjects(PathHierarchyPaintingHelper.java:160)
    at qupath.lib.gui.viewer.overlays.HierarchyOverlay.paintOverlay(HierarchyOverlay.java:231)
    at qupath.lib.gui.viewer.QuPathViewer.paintViewer(QuPathViewer.java:1761)
    at qupath.lib.gui.viewer.QuPathViewer.paintCanvas(QuPathViewer.java:420)
    at qupath.lib.gui.viewer.QuPathViewerPlus.paintCanvas(QuPathViewerPlus.java:257)
    at qupath.lib.gui.viewer.QuPathViewer.lambda$repaint$4(QuPathViewer.java:508)
    at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
...

**Comment by petebankhead** [comment]
Thanks @ozen I can replicate the issue with this file. The error message shows that the problem comes as QuPath tries to draw the annotation on screen at a low resolution.

If shouldn't affect files without annotations, but I should be able to take steps to make this bit of the code more robust.

In the meantime, you can try this script to update the ROIs to the 'new' format. That appears to resolve the problem for me.
```groovy
def annotations = getAnnotationObjects()
annotations = annotations.each { a ->
    if (a.getROI().getClass().getSimpleName().contains('Area'))
        a.setROI(GeometryTools.geometryToROI(a.getROI().getGeometry(), a.getROI().getImagePlane()))
}
fireHierarchyUpdate()
```

I think (but haven't confirmed) that it should be possible to use _Run &rarr; Run for project_ to apply this across your project (but of course best to do it to a copy, not the original!).

**Comment by petebankhead** [comment]
Addressed in v0.2.0-m7 (viewer should no longer crash if some a ROI is encountered again).
